nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BatchUploadObject.java

```diff
@@ -104,6 +110,78 @@
 
     public static final String UPLOAD_TYPE_CHUNKED = "chunked";
 
+    private BatchManager batchManager;
+
+
+    @Override protected void initialize(Object... args) {
+        super.initialize(args);
+        batchManager = Framework.getService(BatchManager.class);
+    }
+
+    @GET
+    @Path("handlers")
+    public Response providers() throws IOException {
+        Map<String, BatchHandler> supportedHandlers = batchManager.getSupportedHandlers();
+        Map<String, Object> result = new HashMap<>();
+
+        result.put("handlers",
+                supportedHandlers.entrySet()
+                                 .stream()
+                                 .map(entry -> Collections.singletonMap("name",
+                                         entry.getKey()))
+                                 .collect(Collectors.toList())
+        );
+
+        return buildResponse(Status.OK, result);
+    }
+
+    @GET
+    @Path("handlers/{handlerId}")
+    public Response getProviderInfo(@PathParam("handlerId") String handlerId) throws IOException {
+        BatchHandler handler = batchManager.getHandlerByName(handlerId);
+
+        if (handler == null) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+
+        Map<String, Object> result = new HashMap<>();
+        result.put("name", handler.getName());
+
+        return buildResponse(Status.OK, result);
+    }
+
+    @POST
+    @Path("new/{providerId}")
+    public Response createNewBatch(@Context UriInfo uriInfo, @PathParam("providerId") String providerId) throws IOException {
+        if (StringUtils.isEmpty(providerId)) {
+            String batchId = initBatchInternal();
+            return Response.seeOther(
+                    UriBuilder
+                        .fromResource(BatchUploadObject.class)
+                        .path("/{batchId}")
+                        .build(batchId)
+                    ).build();
+        }
+
+        Batch batch = initBatchInternal(providerId);
+
+        return Response.seeOther(
+                UriBuilder
+                        .fromPath(getContext().head().getPath().replace("/nuxeo/site", StringUtils.EMPTY))
+                        .path("/upload/{batchId}/info")
+                        .build(batch.getKey())
+        ).build();
+    }
+
+
+    private String initBatchInternal() {
+        return batchManager.initBatch();
+    }
+
+    private Batch initBatchInternal(String providerName) {
+        return batchManager.initBatch(providerName);
```

Comment:
Inline these two trivial methods, they don't bring anything.

Reply:
What do you mean by `inline` ?

Reply:
Just let the base method call `batchManager.initBatch(providerName)` directly instead of calling `initBatchInternal(providerName)` that itself just calls `batchManager.initBatch(providerName)`